### PR TITLE
Upload images

### PR DESCRIPTION
**Pull request by Brianna-Flo**
## Description

<what this pull request is doing>

- integrate the imgbb api to allow users to upload images when making edits to a recipe
- Display the uploaded image after submitting the chosen file
- Display the uploaded image across other accounts after submission

<what will be done in later PRs and not included here>

## Milestones

<the milestones or stories/features that this works towards>

- user personalization

## Resources

<links to tutorials, code snippets, inspirations>
<if you took or translated specific code from the internet, please call it out here!>
https://api.imgbb.com/
https://stackoverflow.com/questions/74819349/when-i-am-trying-to-upload-image-on-imgbb-from-my-react-app-by-fetch-i-am-gettin

## Test Plan

<insert images or gifs of feature>
<otherwise, say how code was tested if not UI facing>
<any edge cases you might have specifically tested>
<div>
    <a href="https://www.loom.com/share/e0ae3cce33f540d79bc35641fa3065e8">
      <p>Image Upload Demo - Watch Video</p>
    </a>
    <a href="https://www.loom.com/share/e0ae3cce33f540d79bc35641fa3065e8">
      
    </a>
  </div>

### CODE REVIEW

File: frontend/src/components/recipeDiff/ImageSearchModal.tsx [hunk 1]
```diff
@@ -120,11 +121,31 @@ const ImageSearchModal = ({
       <ModalDialog sx={GPModalStyle}>
         <ModalClose variant="plain" sx={{ zIndex: 2, m: 1 }} />
         <DialogContent sx={{ my: 3 }}>
+          <FormControl>
+            <FormLabel>Add your own image!</FormLabel>
+            <Input
+              sx={{
+                border: "none",
```
Comment (paulshaoyuqiao) [inline frontend/src/components/recipeDiff/ImageSearchModal.tsx:128]: nit: this can probably be declared up front?

**Review by paulshaoyuqiao — APPROVED** [review]
